public/app/core/components/SharedPreferences/SharedPreferences.tsx

```diff
@@ -95,44 +112,44 @@ export class SharedPreferences extends PureComponent<Props, State> {
   async componentDidMount() {
     const prefs = await this.service.load();
     const dashboards = await backendSrv.search({ starred: true });
-    const defaultDashboardHit: DashboardSearchHit = {
-      id: 0,
-      title: 'Default',
-      tags: [],
-      type: '' as DashboardSearchItemType,
-      uid: '',
-      uri: '',
-      url: '',
-      folderId: 0,
-      folderTitle: '',
-      folderUid: '',
-      folderUrl: '',
-      isStarred: false,
-      slug: '',
-      items: [],
-    };
 
-    if (prefs.homeDashboardId > 0 && !dashboards.find((d) => d.id === prefs.homeDashboardId)) {
-      const missing = await backendSrv.search({ dashboardIds: [prefs.homeDashboardId] });
-      if (missing && missing.length > 0) {
-        dashboards.push(missing[0]);
+    if (prefs.homeDashboardUID && !dashboards.find((d) => d.uid === prefs.homeDashboardUID)) {
+      const missingDash = await backendSrv.getDashboardByUid(prefs.homeDashboardUID);
+
+      if (missingDash?.dashboard) {
+        dashboards.push({
+          id: missingDash.dashboard.id,
```

Comment:
We're using here the ID because it is part of the `DashboardSearchHit` which is the type accepted by select. 

Since we want to remove all uses of ID, we have different options:
* The select do not use it to pick a value, so maybe we can set it to `0` and it will keep working. 
* Wait until we complete the epic to remove the required `id` for this type to stop using it all in `DashboardSearchHit`
* Use a different type for the select

I'd suggest waiting until we progress more with the epic, to remove it everywhere and keep the scope of this PR narrowed to preferences. WDYT?